lib/fastly_nsq/consumer.rb

```diff
@@ -11,13 +11,13 @@ class Consumer
     def initialize(topic:, channel:, ssl_context: nil, connector: nil)
       @topic       = topic
       @channel     = channel
-      @ssl_context = SSLContext.new(ssl_context)
+      @tls_options = TlsOptions.new(ssl_context).to_h
```

Comment:
Seems a little odd that you change `SslContext` to `TlsOptions` but left `ssl_context` in the method signature. I know that means less downstream changes but it definitely is a bit confusing.

I also think the `to_h` should be inside the merge, but I don't really have a great reason why.